waku/v2/rpc/waku_rpc.go

```diff
@@ -72,15 +73,20 @@ func NewWakuRpc(node *node.WakuNode, address string, port int) *WakuRpc {
 	}
 
 	return &WakuRpc{
-		node:         node,
-		server:       server,
-		relayService: relayService,
+		node:          node,
+		server:        server,
+		relayService:  relayService,
+		filterService: filterService,
 	}
 }
 
 func (r *WakuRpc) Start() {
 	go r.relayService.Start()
 	defer r.relayService.Stop()
+
+	go r.filterService.Start()
+	defer r.filterService.Stop()
+
 	log.Info("Rpc server started at ", r.server.Addr)
 	log.Info("server stopped ", r.server.ListenAndServe())
 }
```

Comment:
Should the `r.__Service.Stop()` functios be called in a separate `func(r *WakuRpc) Stop()` function that is invoked in node.go just before `wakuNode.Stop()` ?

Reply:
The start of the RPC server is trigger via go coroutine, this call:
```
 	log.Info("server stopped ", r.server.ListenAndServe()) 
``
Is going to block it until it is stopped, so it is working I think as expected but maybe I can make it better with a proper stop and replace the ListenAndServe with something non blocking. What do u prefer?

Reply:
@richard-ramos let me know if you prefer the new implementation

Reply:
Looks cool!